diff --git a/binsrc/fct/rdfdesc/description.vsp b/binsrc/fct/rdfdesc/description.vsp
index 14c7855b6..58a92680d 100644
--- a/binsrc/fct/rdfdesc/description.vsp
+++ b/binsrc/fct/rdfdesc/description.vsp
@@ -147,7 +147,7 @@ again:
   langs := http_request_header_full (lines, 'Accept-Language', 'en');
   ua :=   http_request_header (lines, 'User-Agent');
   all_langs := b3s_get_lang_acc (lines);
-  lang_parm := get_keyword ('lang', params, '');
+  lang_parm := get_keyword ('lang', params, '*');
   if (length (lang_parm))
     {
       all_langs := vector (lang_parm, 1.0);
diff --git a/libsrc/Wi/sparql_io.sql b/libsrc/Wi/sparql_io.sql
index 3169b5aa4..900fb2bde 100644
--- a/libsrc/Wi/sparql_io.sql
+++ b/libsrc/Wi/sparql_io.sql
@@ -3282,7 +3282,7 @@ create procedure WS.WS."/!sparql/" (inout path varchar, inout params any, inout
   named_graphs := vector ();
   using_graphs := vector ();
   using_named_graphs := vector ();
-  maxrows := 1024*1024; -- More than enough for web-interface.
+  maxrows := ((16*1024*1024)/8)-2; -- More than enough for web-interface.
   deadl := 0;
   http_meth := http_request_get ('REQUEST_METHOD');
   ini_dflt_graph := virtuoso_ini_item_value ('SPARQL', 'DefaultGraph');
